cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/workflow/BasicWorkflowToken.java

```diff
@@ -37,14 +38,24 @@
 
   private static final long serialVersionUID = -1173500180640174909L;
 
-  private Map<String, Map<String, Long>> mapReduceCounters;
   private final Map<Scope, Map<String, List<NodeValue>>> tokenValueMap = new EnumMap<>(Scope.class);
+  private final int maxSizeBytes;
+  private Map<String, Map<String, Long>> mapReduceCounters;
   private String nodeName;
+  private int currentSizeBytes;
```

Comment:
Why do we store the size in MDS? Seems like this field should be transient. Actually only the tokenValueMap should be saved, isn't it (or may the mapReduceCounters as well).